basecoin/store/src/impls/in_memory.rs

```diff
@@ -66,11 +114,29 @@ impl Store for InMemoryStore {
     }
 
     fn commit(&mut self) -> Result<Vec<u8>, Self::Error> {
-        trace!("committing height: {}", self.store.len());
-        self.store.push(self.pending.clone());
+        self.apply()?;
+        trace!("committing height: {}", self.current_height());
+        self.store.push(self.staged.clone());
         Ok(self.root_hash())
     }
 
+    fn apply(&mut self) -> Result<(), Self::Error> {
+        trace!("applying height: {}", self.current_height());
+        self.staged = self.pending.clone();
+        Ok(())
+    }
+
+    fn reset(&mut self) {
+        trace!("resetting height: {}", self.current_height());
+        self.pending = self.staged.clone();
+    }
+
+    fn prune(&mut self, height: RawHeight) -> Result<RawHeight, Self::Error> {
```

Comment:
Having this method return the `RawHeight` seems a bit weird, since the caller will already have access to it. Would it be better to have it instead return `Result<(), Self::Error>`?

Reply:
Spot on. The `Store` trait methods need improvement. Refactoring this here would require changes in all implementations.